web/js/panel.js

```diff
@@ -1196,11 +1196,13 @@ function processPanelXML($returnedData, $success, $xhr) {
                             //draw the LayoutShape
                             $drawLayoutShape($widget);
                             break;
+                        case "positionableRectangle" : //just like RoundRect except cornerRadius set to 0;
                         case "positionableRoundRect" :
                             //log.log("#### positionableRoundRect ####");
                             //copy and reformat some attributes from children into object
                             $widget['width'] = $(this).find('size').attr('width');
                             $widget['height'] = $(this).find('size').attr('height');
+                            $widget['cornerRadius'] = 0; //default to no corner
                             $widget['cornerRadius'] = $(this).find('size').attr('cornerRadius');
```

Comment:
Is the first line needed? The second line always sets the radius unless an exception happens. Unless there is something with JavaScript that I don't understand.

Reply:
If an exception is thrown here, will the rest of the method be executed? Will the line color, fill color, and other stuff be set?

An other solution to this problem is to add `cornerRadius` to `positionableRectangle`. You can add a comment that it's used by the panel file. To add it, you need to:

- Add `<xs:attribute name="cornerRadius" type="xs:int"></xs:attribute>` to `xml/schema/types/editors.xsd` at line 412. See `EditorPositionableRectangleType` and `EditorPositionableRoundRectType`.

- Add `elem.setAttribute("cornerRadius", "0");` to `jmri/jmrit/display/controlPanelEditor/shape/configurexml/PositionableRectangleXml.java` at line 39. See `jmri/jmrit/display/controlPanelEditor/shape/configurexml/PositionableRoundRectXml.java`.

Reply:
Yes, the remainder of the attributes are processed.

What happens is nothing like an exception in Java. 
Basically, the 2nd line fails quietly, and does not set the variable, and the code continues on to the next line. No error is logged.

As far as changing the incoming XML, I have no objection to that, but I don't normally expand the solution to that level. I just deal with it in panel.js and move on.
In this particular case, I wondered why even have separate types with only that one difference. But that's chasing a rabbit that I don't need to chase.

Reply:
I see. Thanks for explaining it!